landlab/utils/tests/test_stream_length.py

```diff
@@ -0,0 +1,156 @@
+    assert_array_equal(flow_length_expected, flow_length)
+
+
+def test_stream_length_irregular_grid_d4():
+    """Test to demonstrate that stream_length utility works as expected with irregular grids"""
+    # instantiate a model grid
+    dx=(2./(3.**0.5))**0.5
+    hmg = HexModelGrid(5,3, dx)
+    # instantiate and add the elevation field
+    hmg.add_field('topographic__elevation', hmg.node_x + np.round(hmg.node_y), at = 'node')
+    # instantiate the expected flow_length array
+    flow_length_expected = np.array ([0,0,0,0,0,dx,0,0,dx,2*dx,2*dx,0,0,2*dx,3*dx,0,0,0,0])
+    #setting boundary conditions   
+    hmg.status_at_node[0] = CLOSED_BOUNDARY
+    # calculating flow directions with FlowAccumulator component: D4 algorithm
+    fr = FlowAccumulator(hmg, flow_director = 'D4')
+    fr.run_one_step()
+    # calculating flow length map
+    flow_length = calculate_stream_length(hmg, add_to_grid=True, noclobber=False)
+    # modifying the stream_length_map because boundary and outlet nodes should not
+    # have stream_length value different from 0
+    flow_length[hmg.boundary_nodes] = 0
+    outlet_id = 4
+    flow_length[outlet_id] = 0
+    hmg.add_field('node','flow_length',flow_length)
+    # test that the stream length utility works as expected
+    assert_array_equal(flow_length_expected, flow_length)
+
+
+def test_stream_length_raster_MFD():
+    """Test of stream length utility with a raster grid and MFD."""
+    # instantiate a model grid
+    mg = RasterModelGrid((5, 4), spacing=(1, 1))
+    # instantiate an elevation array
+    z = np.array ([[0,0,0,0],[0,21,10,0],[0,31,20,0],[0,32,30,0],[0,0,0,0]],dtype='float64')
+    # add the elevation field to the grid
+    mg.add_field('node','topographic__elevation', z)
+    # instantiate the expected flow_length array
+    # considering flow directions calculated with MFD algorithm
+    flow_length_expected = np.array ([[0,0,0,0],[0,1,0,0],[0,math.sqrt(2),1,0],[0,1+math.sqrt(2),2,0],[0,0,0,0]],dtype='float64')
+    flow_length_expected = np.reshape (flow_length_expected,mg.number_of_node_rows*mg.number_of_node_columns)
+    #setting boundary conditions
+    mg.set_closed_boundaries_at_grid_edges(bottom_is_closed=True, left_is_closed=True, right_is_closed=True, top_is_closed=True)
+    # calculating flow directions with FlowAccumulator component
+    fa = FlowAccumulator(mg, 'topographic__elevation',flow_director='MFD',diagonals = True)
+    #If diagonals = False, the flow_length_expected becomes:
+    #flow_length_expected = np.array ([[0,0,0,0],[0,1,0,0],[0,2,1,0],[0,3,2,0],[0,0,0,0]],dtype='float64')
+    fa.run_one_step()
+    # calculating flow length map
+    flow_length = calculate_stream_length(mg, add_to_grid=True, noclobber=False)
+    # test that the stream length utility works as expected
+    assert_array_equal(flow_length_expected, flow_length)
+
+
+def test_stream_length_raster_D_infinity():
+    """Test of stream length utility with a raster grid and D infinity."""
```

Comment:
@Giuseppecipolla95 I think you should be able to create an expected stream length field for this. 

The issue you are having has to do with the fact that the grid you have created results in all of the nodes being assigned as flowing to themselves with DINF. I'll have to take a more close look and identify if this is a bug in D infinity or if it is behaving correctly. 

This will almost certainly happen after next Wednesday.

Reply:
@kbarnhart  Ok, I'm waiting for your response.